evap/staff/views.py

```diff
@@ -691,31 +691,22 @@ def course_person_import(request, semester_id, course_id):
             if excel_form.is_valid():
                 excel_file = excel_form.cleaned_data['excel_file']
                 file_content = excel_file.read()
-                # if on a test run, the process method will not return a list of users that would be
-                # imported so there is currently no way to show how many users would be imported.
-                __, success_messages, warnings, errors = UserImporter.process(file_content, test_run=True)
+                success_messages, warnings, errors = PersonImporter.process(import_type, course, test_run=True, file_content=file_content)
                 if not errors:
                     save_import_file(excel_file, request.user.id, import_type)
 
         elif 'import' in operation:
             file_content = get_import_file_content_or_raise(request.user.id, import_type)
-            imported_users, success_messages, warnings, __ = UserImporter.process(file_content, test_run=False)
+            success_messages, warnings, __ = PersonImporter.process(import_type, course, test_run=False, file_content=file_content)
             delete_import_file(request.user.id, import_type)
-            # Import happens in this call:
-            success_messages.append(helper_person_import(imported_users, course, import_type))
             forward_messages(request, success_messages, warnings)
             return redirect('staff:semester_view', semester_id)
 
         elif 'copy' in operation:
             copy_form.course_selection_required = True
             if copy_form.is_valid():
                 import_course = copy_form.cleaned_data['course']
-                if import_type == 'participant':
-                    imported_users = import_course.participants.all()
-                else:
-                    imported_users = UserProfile.objects.filter(contributions__course=import_course)
-                # Import happens in this call:
-                success_messages.append(helper_person_import(imported_users, course, import_type))
+                success_messages, warnings, errors = PersonImporter.process(import_type, course, test_run=False, source_course=import_course)
```

Comment:
i like this very much. the view definitely had too much going on so moving these out of it is a good thing in my opinion.